Testscripts/Windows/STRESSTEST-VERIFY-RESTART.ps1

```diff
@@ -17,54 +17,107 @@ function Main {
+            break
+        }
+        {($_ -imatch "SRIOV")} {
+            $keyWord = "Mellanox"
+            foreach($nic in (Get-AzVM -Name $AllVmData.RoleName).NetworkProfile.NetworkInterfaces) {
+                if((Get-AzNetworkInterface -ResourceId $nic.Id).EnableAcceleratedNetworking) {
+                    $expectedcount++
+                }
             }
-            $resultArr += $testResult
+            break
         }
+        default { }
     }
-    $rebootNr--
-    Write-LogInfo "Reboot Stress Test Result: $rebootNr/$rebootNumber"
-    if (($rebootNr - 1) -lt $rebootNumber) {
+    Run-LinuxCmd -ip $AllVmData.PublicIP -port $AllVmData.SSHPort -username $user -password $password -command "which lspci || (. ./utils.sh && install_package pciutils)" -runAsSudo | Out-Null
+    $noIssues = Check-KernelLogs -allVMData $AllVmData
+    if (!$noIssues) {
+        $CurrentTestResult.TestSummary += New-ResultSummary -testResult "FAIL" `
+            -metaData "FirstBoot : Call Trace Verification" -checkValues "PASS,FAIL,ABORTED" -testName $currentTestData.testName
         $testResult = "FAIL"
+    } else {
+        for ($rebootNr = 1; $rebootNr -le $rebootNumber; $rebootNr++) {
+            try {
+                Write-LogInfo ("Trying to restart {0}: {1} / {2} ..." `
+                    -f @($AllVMData.RoleName, $rebootNr, $rebootNumber))
+                $isRestarted = $TestProvider.RestartAllDeployments($AllVmData)
+                if ($isRestarted) {
+                    Write-LogInfo "Virtual machine restart successful."
+                    $noIssues = Check-KernelLogs -allVMData $AllVmData
+                    if (!$noIssues) {
+                        $CurrentTestResult.TestSummary += New-ResultSummary -testResult "FAIL" `
+                            -metaData "The $rebootNr Reboot: Call Trace Verification" -checkValues "PASS,FAIL,ABORTED" -testName $currentTestData.testName
+                        $testResult = "FAIL"
+                        break
+                    } else {
+                        if ($testType) {
+                            $pciDevices = Run-LinuxCmd -ip $AllVmData.PublicIP -port $AllVmData.SSHPort -username $user -password $password -command "lspci | grep -i '$keyWord' | wc -l" -runAsSudo -ignoreLinuxExitCode
+                            Write-LogInfo "Actual count: $pciDevices, expected count: $expectedcount"
+                            if ($pciDevices -ne $expectedcount) {
+                                $CurrentTestResult.TestSummary += New-ResultSummary -testResult "FAIL" `
+                                    -metaData "The $rebootNr Reboot: lspci return values are not expected" -checkValues "PASS,FAIL,ABORTED" -testName $currentTestData.testName
+                                $testResult = "FAIL"
+                                break
```

Comment:
do we really want 'break' from all result collection, is it better to collect all of those Failure reason from all restart iterations?

Reply:
When issue happen, we want to keep the site, and see what happen at that time?

Reply:
> do we really want 'break' from all result collection, is it better to collect all of those Failure reason from all restart iterations?

When issue happen, we want to keep the site, and see what happen at that time?

Reply:
it's ok if our design just wants to catch the first failure during rebooting and deployments, break and skip others.  If our design is try to see if another reboot can overcome some random errors, then use continue.  I'm not stick to testing logics. both seems ok.